src/Arcus.EventGrid/EventGridMessage.cs

```diff
@@ -28,41 +25,6 @@ public EventGridMessage(string sessionId)
         /// <summary>
         ///     List of all events, belonging to the Event Grid message
         /// </summary>
-        public List<Event<TData>> Events { get; } = new List<Event<TData>>();
-
-        /// <summary>
-        ///     Parses a string to a EventGridMessage with typed data payload
-        /// </summary>
-        /// <param name="rawJsonBody">Raw JSON body</param>
-        /// <returns>Typed EventGridMessage</returns>
-        public static EventGridMessage<TData> Parse(string rawJsonBody)
-        {
-            Guard.NotNullOrWhitespace(rawJsonBody, nameof(rawJsonBody));
-            var sessionId = Guid.NewGuid().ToString();
-
-            var eventGridMessage = Parse(rawJsonBody, sessionId);
-            return eventGridMessage;
-        }
-
-        /// <summary>
-        ///     Parses a string to a EventGridMessage with typed data payload
-        /// </summary>
-        /// <param name="rawJsonBody">Raw JSON body</param>
-        /// <param name="sessionId">Session id for event grid message</param>
-        /// <returns>Typed EventGridMessage</returns>
-        public static EventGridMessage<TData> Parse(string rawJsonBody, string sessionId)
-        {
-            var array = JArray.Parse(rawJsonBody);
-            var result = new EventGridMessage<TData>(sessionId);
-
-            foreach (var eventObject in array.Children<JObject>())
-            {
-                var rawEvent = eventObject.ToString();
-                var gridEvent = JsonConvert.DeserializeObject<Event<TData>>(rawEvent);
-                result.Events.Add(gridEvent);
-            }
-
-            return result;
-        }
+        public List<TEvent> Events { get; } = new List<TEvent>();
```

Comment:
Exposing a sequence object entirly in such a matter is a bit bad practice.
If this sequence should not be altered afterwards, expose it as `IEnumerable<>` and call the `.AsEnumerable()` on the list type. (Or it could even be a `Collection` if we only add items).

Reply:
Yeah, if you feel more comfortable with that; could also work; but either way without exposing the manipulation of the sequence. In the context of "Only expose what you need".

Reply:
Ok for you @samneirinck?

Reply:
Now, only this and it's OK I think.

Reply:
Done!